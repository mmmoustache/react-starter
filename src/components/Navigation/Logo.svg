<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 15.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="933.333px" height="933.333px" viewBox="0 0 933.333 933.333" enable-background="new 0 0 933.333 933.333"
	 xml:space="preserve">
<path d="M233.066,233.467L0,466.667L117.066,583.6c64.267,64.4,169.333,169.467,233.2,233.467l116.4,116.266L700,700
	l233.332-233.333L700,233.333C571.733,105.066,466.533,0,466.399,0.133C466.133,0.133,361.2,105.2,233.066,233.467L233.066,233.467z
	"/>
</svg>
