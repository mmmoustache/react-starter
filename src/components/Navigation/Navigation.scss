@import '../../styles/settings.scss';
@import '../../styles/tools.scss';

.root {
  background: t-colour(Navigation, Background);
  border-bottom: 1px solid t-colour(General, Border);
  height: t-rem(60px);
  overflow: hidden;
  position: relative;
}

.link {
  color: t-colour(General, Primary);
  display: block;
  padding: t-rem(18px);
  text-decoration: none;

  &:hover,
  &:active,
  &:focus {
    color: t-colour(General, Primary);
    text-decoration: underline;
  }
}

.list {
  @include t-layer(first);

  background: t-colour(Navigation, Mobile);
  float: left;
  height: 100%;
  left: 0;
  overflow: hidden;
  position: fixed;
  width: 100%;

  @include t-breakpoint-min (medium) {
    background: none;
    position: relative;
    width: auto;
  }
}

.listHidden {
  composes: list;
  display: none;

  @include t-breakpoint-min (medium) {
    display: block;
  }
}

.listInner {
  @include t-centerer;
  @include t-reset-list;

  overflow: hidden;
  width: 100%;

  @include t-breakpoint-min (medium) {
    left: 0;
    position: relative;
    top: 0;
    transform: none;
  }
}

.listItem {
  text-align: center;
  width: 100%;

  @include t-breakpoint-min (medium) {
    float: left;
    width: auto;
  }
}

.logo {
  @include t-layer(second);

  color: t-colour(General, Primary);
  display: block;
  float: left;
  margin-right: t-rem(20px);
  position: relative;
}

.logoImage {
  display: block;
  float: left;
  height: t-rem(40px);
  width: t-rem(40px);
  margin: t-rem(10px 0);
}

.logoText {
  @include CopyBold;

  display: block;
  float: left;
  padding: t-rem(18px 18px 18px 10px);
}

.toggle {
  @include t-layer(second);

  cursor: pointer;
  height: 100%;
  position: absolute;
  right: 0;
  text-indent: t-rem(-9999px);
  top: 0;
  width: t-rem(54px);

  @include t-breakpoint-min (medium) {
    display: none;
  }
}

.toggleInner {
  @include t-centerer;
  @include t-transition(short, default, background);

  background: t-colour(General, Primary);
  display: block;
  height: t-rem(1px);
  width: t-rem(20px);

  &::before {
    @include t-toggleLine;
  }

  &::after {
    @include t-toggleLine;

    top: t-rem(6px);
  }
}

.toggleInnerActive {
  composes: toggleInner;
  background: transparent;

  &::before {
    top: auto;
    transform: rotate(-45deg);
  }

  &::after {
    top: 0;
    transform: rotate(45deg);
  }
}
