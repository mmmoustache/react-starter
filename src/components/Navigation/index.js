import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import s from './Navigation.scss';
import routes from '../../routes';
import Logo from './Logo.svg';
import Row from '../Row';
import Column from '../Column';

class Navigation extends React.Component {
  render() {
    const { navigationIsHidden, handleToggle, isMobile } = this.props;
    return (
			<header className={s.root}>
        <Row className="small-collapse large-uncollapse">
          <Column className="small-12">
            <Link to="/" className={s.logo} title="React Starter" onClick={isMobile ? handleToggle : () => {}}>
              <img src={Logo} alt="React Starter" className={s.logoImage} />
              <span className={s.logoText}>React Starter</span>
            </Link>
            <nav className={navigationIsHidden ? s.listHidden : s.list}>
              <ul className={s.listInner}>
                {
                  routes.filter(e => e.showInNav).map(e => 
                    <li key={e.path} className={s.listItem}>
                      <Link to={e.path} className={s.link} title={e.title} onClick={isMobile ? handleToggle : () => {}}>{e.title}</Link>
                    </li>
                  )
                }
              </ul>
            </nav>
            <button type="button" onClick={isMobile ? handleToggle : () => {}} className={s.toggle}>
              <span className={navigationIsHidden ? s.toggleInner : s.toggleInnerActive}>
                Menu
              </span>
            </button>
          </Column>
        </Row>
			</header>
		);
	}
}

Navigation.propTypes = {
  navigationIsHidden: PropTypes.bool.isRequired,
  handleToggle: PropTypes.func.isRequired,
  isMobile: PropTypes.bool.isRequired,
};

export default Navigation;
