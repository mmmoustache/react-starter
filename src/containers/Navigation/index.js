import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Navigation from '../../components/Navigation';
import updateNavVisibility from '../../actions/updateNavVisibility';

class NavigationComponent extends React.Component {

  static propTypes = {
    updateNavVisibility: PropTypes.func.isRequired,
    navigationIsHidden: PropTypes.bool.isRequired,
  };

	constructor() {
		super();

    this.handleResize = this.handleResize.bind(this);
    this.handleToggle = this.handleToggle.bind(this);
  }

  componentWillMount() {
    this.handleResize();
  }

	componentDidMount() {
    window.addEventListener('resize', this.handleResize);
	}

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  }

  handleResize() {
    this.setState({
      isMobile: window.innerWidth < 768,
    });
  }

  handleToggle() {
    this.props.updateNavVisibility(this.props.navigationIsHidden);
  }

	render() {
    const { navigationIsHidden } = this.props;
    return (
      <Navigation 
        isMobile={this.state.isMobile}
        navigationIsHidden={navigationIsHidden} 
        handleToggle={this.handleToggle} />
    );
  }
}

function mapStateToProps(state) {
	return {
		navigationIsHidden: !state.navigationIsHidden,
	};
}

function matchDispatchToProps(dispatch) {
	return bindActionCreators({
		updateNavVisibility: updateNavVisibility,
	}, dispatch);
}

export default connect(mapStateToProps, matchDispatchToProps, null, { pure: false })(NavigationComponent);
