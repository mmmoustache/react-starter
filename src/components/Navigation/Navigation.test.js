import React from 'react';
import Navigation from '../Navigation';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const store = configureStore([
    thunk,
])();

configure({ adapter: new Adapter() });
describe('Component: ContentWrapper', () => {
      
  const component = shallow(
    <Navigation navigationIsHidden isMobile handleToggle={() => {}} store={store} />
  );

  it('Should render successfully', () => {
    expect(component).toMatchSnapshot();
  });
});