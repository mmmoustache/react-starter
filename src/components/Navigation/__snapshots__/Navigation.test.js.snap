// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ContentWrapper Should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    handleToggle={[Function]}
    isMobile={true}
    navigationIsHidden={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row>
        <Column
          small={12}
        >
          <Link
            className={undefined}
            onClick={[Function]}
            replace={false}
            title="React Starter"
            to="/"
          >
            <img
              alt="React Starter"
              className={undefined}
              src=""
            />
            <span
              className={undefined}
            >
              React Starter
            </span>
          </Link>
          <button
            className={undefined}
            onClick={[Function]}
            type="button"
          >
            <span
              className={undefined}
            >
              Menu
            </span>
          </button>
          <nav
            className={undefined}
          >
            <ul
              className={undefined}
            >
              <li
                className={undefined}
              >
                <Link
                  className={undefined}
                  onClick={[Function]}
                  replace={false}
                  title="About"
                  to="/about/"
                >
                  About
                </Link>
              </li>
              <li
                className={undefined}
              >
                <Link
                  className={undefined}
                  onClick={[Function]}
                  replace={false}
                  title="Contact"
                  to="/contact/"
                >
                  Contact
                </Link>
              </li>
            </ul>
          </nav>
        </Column>
      </Row>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Column
          small={12}
        >
          <Link
            className={undefined}
            onClick={[Function]}
            replace={false}
            title="React Starter"
            to="/"
          >
            <img
              alt="React Starter"
              className={undefined}
              src=""
            />
            <span
              className={undefined}
            >
              React Starter
            </span>
          </Link>
          <button
            className={undefined}
            onClick={[Function]}
            type="button"
          >
            <span
              className={undefined}
            >
              Menu
            </span>
          </button>
          <nav
            className={undefined}
          >
            <ul
              className={undefined}
            >
              <li
                className={undefined}
              >
                <Link
                  className={undefined}
                  onClick={[Function]}
                  replace={false}
                  title="About"
                  to="/about/"
                >
                  About
                </Link>
              </li>
              <li
                className={undefined}
              >
                <Link
                  className={undefined}
                  onClick={[Function]}
                  replace={false}
                  title="Contact"
                  to="/contact/"
                >
                  Contact
                </Link>
              </li>
            </ul>
          </nav>
        </Column>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Link
              className={undefined}
              onClick={[Function]}
              replace={false}
              title="React Starter"
              to="/"
            >
              <img
                alt="React Starter"
                className={undefined}
                src=""
              />
              <span
                className={undefined}
              >
                React Starter
              </span>
            </Link>,
            <button
              className={undefined}
              onClick={[Function]}
              type="button"
            >
              <span
                className={undefined}
              >
                Menu
              </span>
            </button>,
            <nav
              className={undefined}
            >
              <ul
                className={undefined}
              >
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="About"
                    to="/about/"
                  >
                    About
                  </Link>
                </li>
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="Contact"
                    to="/contact/"
                  >
                    Contact
                  </Link>
                </li>
              </ul>
            </nav>,
          ],
          "small": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <img
                  alt="React Starter"
                  className={undefined}
                  src=""
                />,
                <span
                  className={undefined}
                >
                  React Starter
                </span>,
              ],
              "className": undefined,
              "onClick": [Function],
              "replace": false,
              "title": "React Starter",
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "React Starter",
                  "className": undefined,
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "React Starter",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "React Starter",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className={undefined}
              >
                Menu
              </span>,
              "className": undefined,
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Menu",
                "className": undefined,
              },
              "ref": null,
              "rendered": "Menu",
              "type": "span",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className={undefined}
              >
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="About"
                    to="/about/"
                  >
                    About
                  </Link>
                </li>
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="Contact"
                    to="/contact/"
                  >
                    Contact
                  </Link>
                </li>
              </ul>,
              "className": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className={undefined}
                  >
                    <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="About"
                      to="/about/"
                    >
                      About
                    </Link>
                  </li>,
                  <li
                    className={undefined}
                  >
                    <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="Contact"
                      to="/contact/"
                    >
                      Contact
                    </Link>
                  </li>,
                ],
                "className": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "/about/",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="About"
                      to="/about/"
                    >
                      About
                    </Link>,
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "About",
                      "className": undefined,
                      "onClick": [Function],
                      "replace": false,
                      "title": "About",
                      "to": "/about/",
                    },
                    "ref": null,
                    "rendered": "About",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "/contact/",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="Contact"
                      to="/contact/"
                    >
                      Contact
                    </Link>,
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Contact",
                      "className": undefined,
                      "onClick": [Function],
                      "replace": false,
                      "title": "Contact",
                      "to": "/contact/",
                    },
                    "ref": null,
                    "rendered": "Contact",
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "nav",
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row>
          <Column
            small={12}
          >
            <Link
              className={undefined}
              onClick={[Function]}
              replace={false}
              title="React Starter"
              to="/"
            >
              <img
                alt="React Starter"
                className={undefined}
                src=""
              />
              <span
                className={undefined}
              >
                React Starter
              </span>
            </Link>
            <button
              className={undefined}
              onClick={[Function]}
              type="button"
            >
              <span
                className={undefined}
              >
                Menu
              </span>
            </button>
            <nav
              className={undefined}
            >
              <ul
                className={undefined}
              >
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="About"
                    to="/about/"
                  >
                    About
                  </Link>
                </li>
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="Contact"
                    to="/contact/"
                  >
                    Contact
                  </Link>
                </li>
              </ul>
            </nav>
          </Column>
        </Row>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Column
            small={12}
          >
            <Link
              className={undefined}
              onClick={[Function]}
              replace={false}
              title="React Starter"
              to="/"
            >
              <img
                alt="React Starter"
                className={undefined}
                src=""
              />
              <span
                className={undefined}
              >
                React Starter
              </span>
            </Link>
            <button
              className={undefined}
              onClick={[Function]}
              type="button"
            >
              <span
                className={undefined}
              >
                Menu
              </span>
            </button>
            <nav
              className={undefined}
            >
              <ul
                className={undefined}
              >
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="About"
                    to="/about/"
                  >
                    About
                  </Link>
                </li>
                <li
                  className={undefined}
                >
                  <Link
                    className={undefined}
                    onClick={[Function]}
                    replace={false}
                    title="Contact"
                    to="/contact/"
                  >
                    Contact
                  </Link>
                </li>
              </ul>
            </nav>
          </Column>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Link
                className={undefined}
                onClick={[Function]}
                replace={false}
                title="React Starter"
                to="/"
              >
                <img
                  alt="React Starter"
                  className={undefined}
                  src=""
                />
                <span
                  className={undefined}
                >
                  React Starter
                </span>
              </Link>,
              <button
                className={undefined}
                onClick={[Function]}
                type="button"
              >
                <span
                  className={undefined}
                >
                  Menu
                </span>
              </button>,
              <nav
                className={undefined}
              >
                <ul
                  className={undefined}
                >
                  <li
                    className={undefined}
                  >
                    <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="About"
                      to="/about/"
                    >
                      About
                    </Link>
                  </li>
                  <li
                    className={undefined}
                  >
                    <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="Contact"
                      to="/contact/"
                    >
                      Contact
                    </Link>
                  </li>
                </ul>
              </nav>,
            ],
            "small": 12,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <img
                    alt="React Starter"
                    className={undefined}
                    src=""
                  />,
                  <span
                    className={undefined}
                  >
                    React Starter
                  </span>,
                ],
                "className": undefined,
                "onClick": [Function],
                "replace": false,
                "title": "React Starter",
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "React Starter",
                    "className": undefined,
                    "src": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "React Starter",
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": "React Starter",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span
                  className={undefined}
                >
                  Menu
                </span>,
                "className": undefined,
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Menu",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "Menu",
                "type": "span",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ul
                  className={undefined}
                >
                  <li
                    className={undefined}
                  >
                    <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="About"
                      to="/about/"
                    >
                      About
                    </Link>
                  </li>
                  <li
                    className={undefined}
                  >
                    <Link
                      className={undefined}
                      onClick={[Function]}
                      replace={false}
                      title="Contact"
                      to="/contact/"
                    >
                      Contact
                    </Link>
                  </li>
                </ul>,
                "className": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <li
                      className={undefined}
                    >
                      <Link
                        className={undefined}
                        onClick={[Function]}
                        replace={false}
                        title="About"
                        to="/about/"
                      >
                        About
                      </Link>
                    </li>,
                    <li
                      className={undefined}
                    >
                      <Link
                        className={undefined}
                        onClick={[Function]}
                        replace={false}
                        title="Contact"
                        to="/contact/"
                      >
                        Contact
                      </Link>
                    </li>,
                  ],
                  "className": undefined,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "/about/",
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className={undefined}
                        onClick={[Function]}
                        replace={false}
                        title="About"
                        to="/about/"
                      >
                        About
                      </Link>,
                      "className": undefined,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "About",
                        "className": undefined,
                        "onClick": [Function],
                        "replace": false,
                        "title": "About",
                        "to": "/about/",
                      },
                      "ref": null,
                      "rendered": "About",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                  Object {
                    "instance": null,
                    "key": "/contact/",
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className={undefined}
                        onClick={[Function]}
                        replace={false}
                        title="Contact"
                        to="/contact/"
                      >
                        Contact
                      </Link>,
                      "className": undefined,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Contact",
                        "className": undefined,
                        "onClick": [Function],
                        "replace": false,
                        "title": "Contact",
                        "to": "/contact/",
                      },
                      "ref": null,
                      "rendered": "Contact",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                ],
                "type": "ul",
              },
              "type": "nav",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
